project/SparkBuild.scala

```diff
@@ -1197,7 +1197,7 @@ object TestSettings {
     (Global / concurrentRestrictions) := {
       // The number of concurrent test groups is empirically chosen based on experience
       // with Jenkins flakiness.
-      if (sys.env.contains("SERIAL_SBT_TESTS")) (Global / concurrentRestrictions).value
+      if (sys.env.contains("SERIAL_SBT_TESTS")) Seq(Tags.limit(Tags.Test, 1))
```

Comment:
Why do we make this change?

Reply:
I'm not sure if `concurrentRestrictions` limits concurrent test to be 1. We can revert this if this is not related.